src/System.Net.Security/src/System/Net/Security/SslServerAuthenticationOptions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    public class SslServerAuthenticationOptions
+    {
+        private X509Certificate _serverCertificate;
+        private X509RevocationMode _checkCertificateRevocation = X509RevocationMode.NoCheck;
+        private SslProtocols _enabledSslProtocols = SecurityProtocol.SystemDefaultSecurityProtocols;
+        private EncryptionPolicy _encryptionPolicy = EncryptionPolicy.RequireEncryption;
+
+        internal RemoteCertValidationCallback _certValidationDelegate;
+        internal LocalCertSelectionCallback _certSelectionDelegate;
+
+        public bool AllowRenegotiation { get; set; }
+
+        public bool ClientCertificateRequired { get; set; }
+
+        public IList<SslApplicationProtocol> ApplicationProtocols { get; set; }
+
+        public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; set; }
+
+        public X509Certificate ServerCertificate
+        {
+            get => _serverCertificate;
+            set => _serverCertificate = value ?? throw new ArgumentNullException(nameof(value));
```

Comment:
`ServerCertificate` starts off as null.

```C#
SslServerAuthenticationOptions options = new SslServerAuthenticationOptions();
// In general, it's weird when a no-op looking thing like this throws.
// In particular, it makes property editors and the debugger do weird things.
options.ServerCertificate = options.ServerCertificate;
```

I'd eliminate the check, or special case it to allow overwriting null with null.

Reply:
ServerCertificate is a required parameter for authentication when run in server mode, there's no default value for this. The above code has incorrect usage.